File: backend/src/main/java/upe/br/ProjetoMentis/business/services/patient/PatientService.java
package upe.br.ProjetoMentis.business.services.patient;

import upe.br.ProjetoMentis.controller.dtos.patient.CreatePatientDto;
import upe.br.ProjetoMentis.controller.dtos.patient.PatientResponseDto;
import upe.br.ProjetoMentis.controller.dtos.patient.UpdatePatientDto;

import java.util.List;
import java.util.UUID;

public interface PatientService {

    PatientResponseDto getPatientById(UUID id);
    List<PatientResponseDto> getAllPatients();
    PatientResponseDto createPatient(CreatePatientDto patient);
    PatientResponseDto updatePatient(UUID id, UpdatePatientDto patient);
    void deletePatient(UUID id);
}

File: frontend/src/app/services/PatientService.ts
import api from './api';
import type { PatientResponse, CreatePatientDto, UpdatePatientDto } from './types';

export async function createPatient(dto: CreatePatientDto): Promise<PatientResponse> {
  const response = await api.post<PatientResponse>('/patients', dto);
  return response.data;
}

export async function getAllPatients(): Promise<PatientResponse[]> {
  const response = await api.get<PatientResponse[]>('/patients');
  return response.data;
}

export async function getPatientById(id: string): Promise<PatientResponse> {
  const response = await api.get<PatientResponse>(`/patients/${id}`);
  return response.data;
}

export async function updatePatient(id: string, dto: UpdatePatientDto): Promise<PatientResponse> {
  const response = await api.put<PatientResponse>(`/patients/${id}`, dto);
  return response.data;
}

export async function deletePatient(id: string): Promise<void> {
  await api.delete(`/patients/${id}`);
}